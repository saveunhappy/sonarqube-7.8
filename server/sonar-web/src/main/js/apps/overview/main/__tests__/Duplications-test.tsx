/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import Duplications from '../Duplications';
import { ComposedProps } from '../enhance';
import {
  mockComponent,
  mockMainBranch,
  mockMeasureEnhanced,
  mockMetric
} from '../../../../helpers/testMocks';

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot();
});

function shallowRender(props: Partial<ComposedProps> = {}) {
  return shallow(
    <Duplications
      branchLike={mockMainBranch()}
      component={mockComponent()}
      leakPeriod={{ index: 1 } as T.Period}
      measures={[
        mockMeasureEnhanced({
          metric: mockMetric({
            id: 'duplicated_lines_density',
            key: 'duplicated_lines_density',
            name: 'Duplicated Lines'
          }),
          value: '0.5'
        }),
        mockMeasureEnhanced({
          metric: mockMetric({
            id: 'new_lines',
            key: 'new_lines',
            name: 'New Lines',
            type: 'INT'
          }),
          value: '52'
        }),
        mockMeasureEnhanced({
          metric: mockMetric({
            id: 'new_duplicated_lines_density',
            key: 'new_duplicated_lines_density',
            name: 'New Duplicated Lines'
          }),
          periods: [
            {
              bestValue: true,
              index: 1,
              value: '1.5'
            }
          ]
        }),
        mockMeasureEnhanced({
          metric: mockMetric({
            id: 'duplicated_blocks',
            key: 'duplicated_blocks',
            name: 'Duplicated Blocks',
            type: 'INT'
          }),
          periods: [
            {
              bestValue: true,
              index: 1,
              value: '2'
            }
          ]
        })
      ]}
      {...props}
    />
  ).dive();
}
