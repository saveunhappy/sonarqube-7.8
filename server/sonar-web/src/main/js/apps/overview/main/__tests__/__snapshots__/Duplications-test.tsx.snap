// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="overview-card"
>
  <div
    className="overview-card-header"
  >
    <div
      className="overview-title"
    >
      <span>
        overview.domain.duplications
      </span>
      <Link
        className="spacer-left small"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/component_measures",
            "query": Object {
              "id": "my-project",
              "metric": "Duplications",
            },
          }
        }
      >
        layout.measures
      </Link>
    </div>
  </div>
  <div
    className="overview-domain-panel"
  >
    <div
      className="overview-domain-nutshell"
    >
      <div
        className="overview-domain-measures overview-domain-measures-big"
      >
        <div
          className="overview-domain-measure"
        >
          <div
            className="display-inline-block text-middle big-spacer-right neg-offset-left"
          >
            <DuplicationsRating
              size="big"
              value={0.5}
            />
          </div>
          <div
            className="display-inline-block text-middle"
          >
            <div
              className="overview-domain-measure-value"
            >
              <DrilldownLink
                branchLike={
                  Object {
                    "analysisDate": "2018-01-01",
                    "isMain": true,
                    "name": "master",
                  }
                }
                component="my-project"
                metric="duplicated_lines_density"
              >
                0.5%
              </DrilldownLink>
            </div>
            <div
              className="overview-domain-measure-label display-flex-center display-flex-justify-center"
            >
              overview.metric.duplications
              <DocTooltip
                className="little-spacer-left"
                doc={Promise {}}
              />
            </div>
            <Link
              className="overview-domain-measure-history-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to={
                Object {
                  "pathname": "/project/activity",
                  "query": Object {
                    "custom_metrics": "duplicated_lines_density",
                    "graph": "custom",
                    "id": "my-project",
                  },
                }
              }
            >
              <HistoryIcon />
              <span>
                project_activity.page
              </span>
            </Link>
          </div>
        </div>
        <div
          className="overview-domain-measure"
        >
          <div
            className="overview-domain-measure-value"
          >
            <DrilldownLink
              branchLike={
                Object {
                  "analysisDate": "2018-01-01",
                  "isMain": true,
                  "name": "master",
                }
              }
              component="my-project"
              metric="duplicated_blocks"
            >
              <span
                className="js-overview-main-tests"
              >
                1
              </span>
            </DrilldownLink>
          </div>
          <div
            className="overview-domain-measure-label display-flex-center display-flex-justify-center"
          >
            Duplicated Blocks
            <DocTooltip
              className="little-spacer-left"
              doc={Promise {}}
            />
            <Link
              className="overview-domain-measure-history-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to={
                Object {
                  "pathname": "/project/activity",
                  "query": Object {
                    "custom_metrics": "duplicated_blocks",
                    "graph": "custom",
                    "id": "my-project",
                  },
                }
              }
            >
              <HistoryIcon />
              <span>
                project_activity.page
              </span>
            </Link>
          </div>
        </div>
      </div>
    </div>
    <div
      className="overview-domain-leak"
    >
      <div
        className="overview-domain-measures"
      >
        <div
          className="overview-domain-measure"
        >
          <div
            className="overview-domain-measure-value"
          >
            <div>
              <DrilldownLink
                branchLike={
                  Object {
                    "analysisDate": "2018-01-01",
                    "isMain": true,
                    "name": "master",
                  }
                }
                component="my-project"
                metric="new_duplicated_lines_density"
              >
                <span
                  className="js-overview-main-new-duplications"
                >
                  1.5%
                </span>
              </DrilldownLink>
            </div>
          </div>
          <div
            className="overview-domain-measure-label"
          >
            overview.duplications_on
            <br />
            <DrilldownLink
              branchLike={
                Object {
                  "analysisDate": "2018-01-01",
                  "isMain": true,
                  "name": "master",
                }
              }
              className="spacer-right overview-domain-secondary-measure-value"
              component="my-project"
              metric="new_lines"
            >
              <span
                className="js-overview-main-new-lines"
              >
                1
              </span>
            </DrilldownLink>
            overview.metric.new_lines
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
